### PR TITLE
Initial near cache prototype

### PR DESCRIPTION
**Pull request by tmiddlet2666**

### CODE REVIEW

**Review by thegridman — DISMISSED** [review]

File: coherence/doc.go [hunk 1]
```diff
@@ -534,6 +535,115 @@ Lastly, when you create a Customer object you must set the Class value matching
 	    log.Fatal(err)
 	}
 
+# Using Near Caches
+
+The Coherence Go client allows you to specify a near cache to cache frequently used and accessed data in your Go application.
```
Comment (rlubke) [inline coherence/doc.go:540]: To me, this wording feels a bit redundant.  Being used implies access, right?

File: coherence/doc.go [hunk 2]
```diff
@@ -534,6 +535,115 @@ Lastly, when you create a Customer object you must set the Class value matching
...
+When you access data using Get() or GetAll() operations, returned entries are stored in the near cache and subsequent data
+access for keys in the near-cache is almost instantaneous.
```
Comment (rlubke) [inline coherence/doc.go:542]: That a near-cache will make data access near instant whereas without a near-cache, each operation results in a network call?

File: coherence/doc.go [hunk 3]
```diff
@@ -534,6 +535,115 @@ Lastly, when you create a Customer object you must set the Class value matching
...
+On creating a near cache, Coherence automatically adds a [MapListener] to your [NamedMap] or [NamedCache] which listens on
+all cache events and updates or invalidates entries in the near cache that have been changed or removed on the server.
+
+To manage the amount of memory used by the near cache, the following options are supported when creating one:
+
+  - time-to-live (TTL) – objects expired after time in near cache, e.g. 5 minutes
+  - High-Units – maximum number of cache entries in the near cache
+  - Memory – maximum amount of memory used by cache entries
+
+Note: You can specify TTL and High-Units or memory.
```
Comment (rlubke) [inline coherence/doc.go:553]: Would it be clearer as "You can specify either High-Units or Memory and in either case, optionally, a TTL"?

File: scripts/run-test-examples.sh [hunk 4]
```diff
@@ -39,6 +39,8 @@ function wait_for_ready() {
 
 wait_for_ready
 
+pause && pause && pause && pause
```
Comment (rlubke) [inline scripts/run-test-examples.sh:42]: What's this?

File: test/e2e/standalone/named_map_test.go [hunk 5]
```diff
@@ -813,6 +813,8 @@ func RunTestGetAll(t *testing.T, namedMap coherence.NamedMap[int, Person]) {
 		results = append(results, se.Key)
 	}
 	g.Expect(len(results)).To(gomega.Equal(0))
+
+	// GetAll keys that don't exist
```
Comment (rlubke) [inline test/e2e/standalone/named_map_test.go:817]: Was this a clarifying comment or something left in by mistake?

File: test/e2e/standalone/suite_test.go [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022, 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2022, 2024 Oracle and/or its affiliates.
```
Comment (rlubke) [inline test/e2e/standalone/suite_test.go:2]: Copyright updated, but no changes.  Was this a post-change copyright update?

**Review by rlubke — APPROVED** [review]
Hi Tim, minor comments.  If you agree with the comments, feel free to make the appropriate changes and submit.

File: coherence/named_cache_client.go [hunk 7]
```diff
@@ -604,6 +649,145 @@ func newNamedCacheReconnectListener[K comparable, V any](nc NamedCacheClient[K,
...
+	if e == Truncated || e == Destroyed {
+		l.Clear()
+	}
+}
+
 func convertNamedCacheClient[K comparable, V any](client *NamedCacheClient[K, V]) NamedMap[K, V] {
 	return client
 }
+
+func validateNearCacheOptions(options *NearCacheOptions) error {
```
[the remark this answers is not in the document]
Comment (tmiddlet2666) [reply coherence/named_cache_client.go:734]: Yes good idea.

File: test/e2e/standalone/suite_test.go (hunk 6 above)
Comment (tmiddlet2666) [reply test/e2e/standalone/suite_test.go:2]: this was a problem when i made a d change then reverted and the copyright plugin didn't like it..

**Review by dhirupandey — APPROVED** [review]

**Pull request merged**